src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -312,6 +315,30 @@ class PythonSetup(Subsystem):
         "on Big Sur.",
     )
 
+    def __init__(self, options: OptionValueContainer) -> None:
+        super().__init__(options)
+        if options.is_default("lockfile_generator"):
+            warn_or_error(
+                "2.12.0.dev0",
```

Comment:
My only concern is the nag starting now. I'd hate to nag to switch from broken to broken if Pex locks come up broken.

Reply:
We aren't yet deprecating Poetry, only relying on Poetry being set as the default. I can reword this to make that more explicit.

Reply:
Ah, we do claim in the message that we'll be changing to Pex in 2.12 as the default. We don't have to stick with that if we aren't ready. We only now have the opportunity to change the default.